src/components/useGetIngredientsIds.tsx

```diff
@@ -0,0 +1,95 @@
+import { NewRecipe, NewIngredient } from "./types.ts";
+import { useMyStore } from "./store.tsx";
+import { fetchPriceAndStock } from "./Api";
+import { useQuery } from "@tanstack/react-query";
+import { useEffect, useState } from "react";
+
+type Product = {
+  id: number;
+  preferred?: boolean | undefined;
+  price: {
+    amount: number;
+  };
+  inStock: boolean;
+};
+
+type IngredientData = null | {
+  productsByStoreId: {
+    [storeId: string]: Product[];
+  };
+  ingredientIds: {
+    [storeId: string]: string[];
+  };
+};
+
+export const useGetIngredientIds = (recipe: NewRecipe) => {
+  const { ingredients } = useMyStore();
+  const ingredientIds = getIngredientIds(recipe, ingredients);
+  const [ingredientData, setIngredientData] = useState<IngredientData | null>(
+    null,
+  );
+
+  const { data } = useQuery(["data", recipe.id], () =>
+    fetchPriceAndStock(ingredientIds),
+  );
+
+  useEffect(() => {
```

Comment:
Tady asi budu potrebovat vysvetleni, co to ma delat. Kazdopadne pro zacatek se muzeme vyhnout tomu mit `ingredientData`  a `setIngredientData`. 

Protoze muzem klidne

```
const { ingredients } = useMyStore();
  const ingredientIds = getIngredientIds(recipe, ingredients);
  const [ingredientData, setIngredientData] = useState<IngredientData | null>(
    null,
  );

  const { data } = useQuery(["data", recipe.id], () =>
   const neco = fetchPriceAndStock(ingredientIds);

   // logika s necimm
  );
  
  return data // nebo cokoli z nich chceme vratit
  
```

Reply:
Teoreticky to vsechno muzem narvat do `useGetRecipePrice`, co je jediny misto, kde to chceme pouzit, ne?

Reply:
super! nemohla jsem prpijit na to, jak to k tomu pridat v ramci te jedne funkce, proto zbytecny stav, abych to jeste doupravila....